ksql-engine/src/test/java/io/confluent/ksql/integration/IntegrationTestHelper.java

```diff
@@ -0,0 +1,480 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.apache.kafka.connect.data.Schema;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+import org.junit.rules.ExternalResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SuppressWarnings("ALL")
+public class IntegrationTestHelper extends ExternalResource implements Closeable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(IntegrationTestHelper.class);
+  private static final int DEFAULT_PARTITION_COUNT = 1;
+  private static final short DEFAULT_REPLICATION_FACTOR = (short) 1;
+  private static final long PRODUCE_TIMEOUT_MS = 30_000;
+
+  private final EmbeddedSingleNodeKafkaCluster kafkaCluster;
+  private final LazyServiceContext serviceContext;
+
+  public IntegrationTestHelper(final EmbeddedSingleNodeKafkaCluster kafkaCluster) {
+    this(kafkaCluster, new MockSchemaRegistryClient());
+  }
+
+  public IntegrationTestHelper(
+      final EmbeddedSingleNodeKafkaCluster kafkaCluster,
+      final SchemaRegistryClient schemaRegistryClient
+  ) {
+    this.kafkaCluster = Objects.requireNonNull(kafkaCluster, "kafkaCluster");
+    this.serviceContext = new LazyServiceContext(kafkaCluster, schemaRegistryClient);
+  }
+
+  /**
+   * Ensure topics with the given {@code topicNames} exist.
+   *
+   * <p>Topics will be creates, if they do not already exist, with a single partition and replica.
+   *
+   * @param topicNames the names of the topics to create.
+   */
+  public void ensureTopics(final String... topicNames) {
+    ensureTopics(DEFAULT_PARTITION_COUNT, topicNames);
+  }
+
+  /**
+   * Ensure topics with the given {@code topicNames} exist.
+   *
+   * <p>Topics will be creates, if they do not already exist, with the specified
+   * {@code partitionCount}.
+   *
+   * @param topicNames the names of the topics to create.
+   */
+  public void ensureTopics(final int partitionCount, final String... topicNames) {
```

Comment:
All of this seems identical to the code in `IntegrationTestHarness`. Is that intentional?

Reply:
Ah, no, this is a `git stash` -> `git stash pop` error, this file shouldn't be included.